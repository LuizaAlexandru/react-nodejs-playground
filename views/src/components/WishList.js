import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import Product from './Product.js';
import {wishlistGetIds, wishlistRemoveProduct} from '../actions/WishlistActions.js';

const mapStateToProps = state => {
    return {
      wishlistProducts: state.user.wishlistProducts
    };
  },
  mapDispatchToProps = dispatch => {
    return {
      wishlistGetIds: () => dispatch(wishlistGetIds()),
      wishlistRemoveProduct: productId => dispatch(wishlistRemoveProduct(productId))
    };
  }
;

class WishList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      userWishlist: [],
      products: []
    }
  }

  componentDidMount() {
    this.props.wishlistGetIds();
  }

  render() {
    return (
      <div className="wishlist-products">
        <h3>WishList</h3>
        <ul className="wishlist-products-list">
          {this.props.wishlistProducts && this.props.wishlistProducts.map(
            (data) => <li className="featured-product-line" key={`product-line-${data._id}`}>
              <Product data={data} actionText={"Remove from wishlist"} callback={this.props.wishlistRemoveProduct}
                actionPlaceholder="- wishlist" mode="simple" /></li>
          )}
        </ul>
      </div>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(WishList);

WishList.propTypes = {
  wishlistProducts: PropTypes.array,
  wishlistGetIds: PropTypes.func,
  wishlistAddProduct: PropTypes.func,
  wishlistRemoveProduct: PropTypes.func,
}
