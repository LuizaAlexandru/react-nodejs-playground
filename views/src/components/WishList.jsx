import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import Product from './Product.jsx';
import Modal from './Modal.jsx';
import Notification from './Notification.jsx';

import { wishlistGetIds, wishlistRemoveProduct } from '../actions/WishlistActions.js';

const mapStateToProps = state => {
    return {
      wishlistProducts: state.user.wishlistProducts,
      wishlistError: state.user.wishlistError
    };
  },
  mapDispatchToProps = dispatch => {
    return {
      wishlistGetIds: () => dispatch(wishlistGetIds()),
      wishlistRemoveProduct: productId => dispatch(wishlistRemoveProduct(productId))
    };
  }
;

class WishList extends Component {

  componentDidMount() {
    this.props.wishlistGetIds();
  }

  render() {
    return (
      <div className="wishlist-products">
        <h3>WishList</h3>
        <ul className="wishlist-products-list">
          {this.props.wishlistProducts && this.props.wishlistProducts.map(
            (data) => <li className="featured-product-line" key={`product-line-${data._id}`}>
              <Product data={data} actionText={"Remove from wishlist"} callback={this.props.wishlistRemoveProduct}
                actionIcon="/imgs/wishlist-delete.svg" mode="simple" /></li>
          )}
        </ul>
        <Modal>
          <Notification icon={'/imgs/warning.svg'} title={this.props.wishlistError}/>
        </Modal>
      </div>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(WishList);

WishList.propTypes = {
  wishlistProducts: PropTypes.array,
  wishlistError: PropTypes.string,
  wishlistGetIds: PropTypes.func,
  wishlistAddProduct: PropTypes.func,
  wishlistRemoveProduct: PropTypes.func,
}
