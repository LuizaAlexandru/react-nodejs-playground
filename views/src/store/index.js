import {createStore, applyMiddleware} from "redux";
import rootReducer from "../reducers/index";

const asyncDispatchMiddleware = store => next => action => {
  let syncActivityFinished = false;
  let actionQueue = [];

  function flushQueue() {
    actionQueue.forEach(a => store.dispatch(a)); // flush queue
    actionQueue = [];
  }

  function asyncDispatch(asyncAction) {
    actionQueue = actionQueue.concat([asyncAction]);

    if (syncActivityFinished) {
      flushQueue();
    }
  }

  const actionWithAsyncDispatch =
      Object.assign({}, action, { asyncDispatch });

  next(actionWithAsyncDispatch);
  syncActivityFinished = true;
  flushQueue();
};

const store = createStore(
  rootReducer,
  applyMiddleware(asyncDispatchMiddleware)
);

export default store;
