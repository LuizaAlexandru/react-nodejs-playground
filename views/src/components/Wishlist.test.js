import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';

import WishList from './Wishlist.js';
import store from '../store/index';

test('Initiates an Wishlist instance', () => {
  const component = renderer.create(
    <Provider store={store}>
      <WishList />
    </Provider>
  );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
