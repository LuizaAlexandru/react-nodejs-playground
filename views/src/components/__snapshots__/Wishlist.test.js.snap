// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initiated Wishlist matches the snapshot 1`] = `
<div
  className="wishlist-products"
>
  <h3>
    WishList
  </h3>
  <ul
    className="wishlist-products-list"
  />
</div>
`;
